### PR TITLE
Added error handling via an alertDialog for 404 and other such errors…

### PR DESCRIPTION
**Pull request by almiche**
[Bug #139] [User Story #117]
AlertDialog

### WHAT kind of change does this PR introduce?
Adding error handling in case the seshat server produces 404 or 400 errors. 

### HOW is this accomplished?
By flashing an alertDialog box at the user alerting them that there has been a problem communicating with the server.

### Checklist
- [ ] Docs have been added / updated to reflect the changes
- [ ] I have reviewed and tested the changes :white_check_mark:
- [ ] I have squashed my commits to have a reasonable amount of commits
- [ ] All of the code I have written adhere to the coding conventions outlined in the coding conventions section of the wiki

### CODE REVIEW

File: app/src/androidTest/java/com/simplemobiletools/calculator/DrawActivityTest.java [hunk 1]
```diff
@@ -28,6 +43,35 @@ public void clearButtonTest() {
...
+        }
+        onWebView().withElement(findElement(Locator.ID, "test")).check(webMatches(getText(), containsString("TEST PAGE")));
+    }
+
+    @Test
+    public void testPageNotFound(){
+        BackgroundGetGraphTaskBuilder backgroundGetGraphTaskBuilder = new BackgroundGetGraphTaskBuilder(activity.getActivity());
+        BackgroundGetGraphTaskBuilder spyBuilder = spy(backgroundGetGraphTaskBuilder);
+        activity.getActivity().supersedeBuilder(spyBuilder);
+        when(spyBuilder.build()).thenReturn(new FakeGraphTask(activity.getActivity(),400));
```
Comment (jusleg) [inline app/src/androidTest/java/com/simplemobiletools/calculator/DrawActivityTest.java]: Nice 400 mocking!

File: app/src/main/assets/succes.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<strong id="test">TEST PAGE</strong>
```
Comment (jusleg) [inline app/src/main/assets/succes.html]: could this file be in the test package? I wouldn't want to mix production assets with test assets if possible

File: app/src/main/kotlin/com/simplemobiletools/calculator/activities/DrawActivity.java [hunk 3]
```diff
@@ -60,6 +69,25 @@ public void supersedeSketchSheetView(SketchSheetView equationSketchView) {
...
+        alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "OK",
+                new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
+                    }
+                });
+        alertDialog.show();
+    }
+
+    public void supersedeBuilder(BackgroundGetGraphTaskBuilder builder) {
```
Comment (jusleg) [inline app/src/main/kotlin/com/simplemobiletools/calculator/activities/DrawActivity.java]: add another `@TestOnly`. I'm bikeshedding 🚲🏠 a bit, but we should make it obvious that it's for testing only and should never be used for prod purpose.

File: app/src/main/assets/succes.html (hunk 2 above)
Comment (almiche) [reply app/src/main/assets/succes.html]: Unfortunately they have to be in the assets folder but I made a test sub-directory within assets to fix it.

File: app/src/androidTest/java/com/simplemobiletools/calculator/DrawActivityTest.java (hunk 1 above)
Comment (almiche) [reply app/src/androidTest/java/com/simplemobiletools/calculator/DrawActivityTest.java]: Thanks!

File: app/src/main/kotlin/com/simplemobiletools/calculator/activities/DrawActivity.java (hunk 3 above)
Comment (almiche) [reply app/src/main/kotlin/com/simplemobiletools/calculator/activities/DrawActivity.java]: 🚲🏠 Bikesheds are cool. Done

File: app/src/main/assets/succes.html (hunk 2 above)
Comment (jusleg) [reply app/src/main/assets/succes.html]: clear enough.

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BackgroundGetGraphTaskBuilder.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.simplemobiletools.calculator.helpers;
+
+import android.os.AsyncTask;
+
+import com.simplemobiletools.calculator.activities.DrawActivity;
+import com.simpletools.calculator.commons.helpers.Calculator;
+
+public class BackgroundGetGraphTaskBuilder {
```
Comment (0matleb2) [inline app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BackgroundGetGraphTaskBuilder.java]: Any particular reason for a Builder pattern here?

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetGraphTask.kt [hunk 5]
```diff
@@ -11,36 +12,52 @@ import java.net.URL
...
+                httpClient.disconnect()
                 return data
-            } catch (e: Exception) {
-                e.printStackTrace()
-            } finally {
+            } else if (httpClient.responseCode == 404) {
                 httpClient.disconnect()
+                pingWithError("Page not found 404!", "There has been a connection problem, returning you to the main page")
+                killThread()
+            } else if (httpClient.responseCode == 400) {
```
Comment (0matleb2) [inline app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetGraphTask.kt]: Could handle all other responseCodes in an else block.

Comment (jusleg) [reply app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetGraphTask.kt]: the server only sends 400 and 404. Maybe if it really fails hard we could get a 500. An `else` would be perfect I think.

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BackgroundGetGraphTaskBuilder.java (hunk 4 above)
Comment (jusleg) [reply app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BackgroundGetGraphTaskBuilder.java]: We used the builder previously to have a way to inject a mock. We reused the same strategy to mock the web request for the graph task

Comment (almiche) [reply app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BackgroundGetGraphTaskBuilder.java]: In order to be able to use a fake a getGraphTask we had to make use of a this builder which can be superseded upon runtime.

**Pull request merged**